File: algorithms/461. Hamming Distance/solution.py
class Solution(object):
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        return bin(x ^ y).count('1')

# 149 / 149 test cases passed.
# Status: Accepted
# Runtime: 48 ms
# 100%

File: algorithms/461. Hamming Distance/solution.cc
class Solution {
public:
    int hammingDistance(int x, int y) {
        // @pengr7
        int z = x ^ y;
        int d = 0;
        while ( z ) {
            d++;
            z &= z - 1;
        }
        return d;
    }
};
